src/mesh/X3D_Draco_Mesh_Reader.cc

```diff
@@ -54,7 +53,10 @@ void X3D_Draco_Mesh_Reader::read_mesh() {
   std::ifstream x3dfile(filename.c_str());
 
   // file must exist and be readable
-  Insist(x3dfile.is_open(), "Failed to find or open specified X3D mesh file.");
+  if (!x3dfile.is_open()) {
+    throw std::runtime_error("Failed to find or open specified X3D mesh file " +
```

Comment:
Insist checks a condition that the client code should always be able to satisfy, so a failure indicates a bug in the code. Successfully opening a file is not something the client can guarantee, because the file system is ultimately not under the control of the client; in other words, it's a legitimate runtime error rather than a bug in the code.